src/main/java/com/basho/riak/client/core/operations/ts/ListKeysOperation.java

```diff
@@ -0,0 +1,97 @@
+package com.basho.riak.client.core.operations.ts;
+
+import com.basho.riak.client.core.operations.PBFutureOperation;
+import com.basho.riak.client.core.query.timeseries.IQueryResult;
+import com.basho.riak.client.core.query.timeseries.immutable.pb.ImmutablePbResultFactory;
+import com.basho.riak.client.core.util.BinaryValue;
+import com.basho.riak.protobuf.RiakMessageCodes;
+import com.basho.riak.protobuf.RiakTsPB;
+import com.google.protobuf.ByteString;
+
+import java.util.List;
+
+/**
+ * Created by alex on 12/3/15.
+ */
+public class ListKeysOperation extends PBFutureOperation<IQueryResult, RiakTsPB.TsListKeysResp, BinaryValue>
+{
+    private final Builder builder;
+    private BinaryValue queryInfoMessage;
+
+    ListKeysOperation(Builder builder)
```

Comment:
Eh, some operations are package local, some are private. Should probably be private though.